### DISCUSSION TITLE
SvelteKit like matchers

### DISCUSSION THREAD

**Discussion by PindaPixel** [posva/unplugin-vue-router#303] [Ideas]
I was playing around with SvelteKit regarding this topic because I stumbled upon a use case for this.
I noticed its imported Params type can be influenced by a matcher's assertion:
![image](https://github.com/posva/unplugin-vue-router/assets/11627777/6f731b56-8257-42ae-a3b9-7951fad73bac)
![image](https://github.com/posva/unplugin-vue-router/assets/11627777/a65420e0-deec-4220-88c2-d5cbe7a4cd08)

Don't know if there are plans for matchers like this, but if there are this would be a cool feature.

**Comment by posva** [comment]
Yes, it’s planned!

**Comment by posva** [comment]
https://github.com/vuejs/router/pull/2415
